src/ripple/app/misc/impl/AmendmentTable.cpp

```diff
@@ -310,9 +319,36 @@ AmendmentTableImpl::AmendmentTableImpl(
     , majorityTime_(majorityTime)
     , unsupportedEnabled_(false)
     , j_(journal)
+    , db_(app.getWalletDB())
 {
     std::lock_guard sl(mutex_);
 
+    // Find out if the FeatureVotes table exists in WalletDB
+    bool FeatureVotesExist;
+    {
+        auto db = db_.checkoutDb();
+        soci::transaction tr(*db);
+        std::string sql =
+            "SELECT count(*) FROM sqlite_master "
+            "WHERE type='table' AND name='FeatureVotes'";
+        boost::optional<int> FeatureVotesCount;
+        *db << sql, soci::into(FeatureVotesCount);
+        FeatureVotesExist = static_cast<bool>(*FeatureVotesCount);
+    }
+
+    // Create FeatureVotes table in WalletDB if it doesn't exist
+    if (!FeatureVotesExist)
+    {
+        auto db = db_.checkoutDb();
+        soci::transaction tr(*db);
+        *db << "CREATE TABLE  FeatureVotes ( "
+               "AmendmentHash      CHARACTER(64) NOT NULL, "
+               "AmendmentName      TEXT, "
+               "Veto               INTEGER NOT NULL );";
+        tr.commit();
+    }
```

Comment:
These two sections can be combined into a single in place lambda, reusing the connection, and allowing `FeatureVotesExist` to be made `const`. I don't think it'll give much performance improvement, but it's a little simpler to read, and logically keeps the two table-creation-related operations together. For example:
```
    bool const featureVotesExist = [&]() {
        auto db = db_.checkoutDb();
        soci::transaction tr(*db);
        std::string sql =
            "SELECT count(*) FROM sqlite_master "
            "WHERE type='table' AND name='FeatureVotes'";
        boost::optional<int> featureVotesCount;
        *db << sql, soci::into(featureVotesCount);
        bool exists = static_cast<bool>(*featureVotesCount);

        // Create FeatureVotes table in WalletDB if it doesn't exist
        if (!exists)
        {
            *db << "CREATE TABLE  FeatureVotes ( "
                   "AmendmentHash      CHARACTER(64) NOT NULL, "
                   "AmendmentName      TEXT, "
                   "Veto               INTEGER NOT NULL );";
            tr.commit();
        }
        return exists;
    }();
```
I also suggest renaming some of the local variables to start with lower case.

Reply:
Done.